### PR TITLE
feat(bootstrap): add kms-key-id option to cdk bootstrap command

### LINKED ISSUE #2203: Add kms encryption key as argument for cdk bootstrap
The `cdk bootstrap` command creates the following resource in CloudFormation:

```
  StagingBucket:
    Type: AWS::S3::Bucket
    Properties:
      AccessControl: Private
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: aws:kms
```

I have not seen any option to specify the KMS key to be used. Is there a way to do so?

Cheers!

### PR DESCRIPTION
**Pull request by abelmokadem**
Closes #2203 

----

### Pull Request Checklist

* [ ] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by abelmokadem** [comment]
Is it possible to see why the build failed?

**Comment by RomainMuller** [comment]
Seems failed due to an ongoing issue we're experiencing with the `mono` tooling when compiling .NET bindings. Ugh. I wonder why you didn't get a TravisCI build here though.

**Comment by abelmokadem** [comment]
Still need to make a change to this code in case there is no key. And add a test case. Haven't had the time yet.

**Comment by RomainMuller** [comment]
That's cool. I was just making sure this PR does grow excessively stale :)

**Comment by RomainMuller** [comment]
@abelmokadem are you till intending to push further commits on there? If you're unable to, it'd be nice if you can lay out  the remaining work you intend to be done before this is merge-able.

**Comment by abelmokadem** [comment]
Hi guys, I have been quite busy with work and our new born daughter :( Sorry I can't pick this up any further. Still hoping to see this feature in a future release.

**Comment by hoegertn** [comment]
Can you outline what is missing? I can try to pick up the PR.

File: packages/aws-cdk/lib/api/bootstrap-environment.ts [hunk 1]
```diff
@@ -7,18 +7,44 @@ import { SDK } from './util/sdk';
...
     template: {
       Description: "The CDK Toolkit Stack. It was created by `cdk bootstrap` and manages resources necessary for managing your Cloud Applications with AWS CDK.",
       Resources: {
         StagingBucket: {
-          Type: "AWS::S3::Bucket",
+          Type: 'AWS::S3::Bucket',
           Properties: {
-            AccessControl: "Private",
-            BucketEncryption: { ServerSideEncryptionConfiguration: [{ ServerSideEncryptionByDefault: { SSEAlgorithm: "aws:kms" } }] }
+            AccessControl: 'Private',
```
Comment (abelmokadem) [inline packages/aws-cdk/lib/api/bootstrap-environment.ts]: @hoegertn there is no condition here for when no `kmsKeyId` is provided.

File: packages/aws-cdk/lib/api/bootstrap-environment.ts [hunk 2]
```diff
@@ -7,18 +7,44 @@ import { SDK } from './util/sdk';
...
+export interface BootstrapEnvironmentProps {
+  /**
+   * The name to be given to the CDK Bootstrap bucket.
+   *
+   * @default a name is generated by CloudFormation.
+   */
+  readonly bucketName?: string;
+
+  /**
+   * The ID of an existing KMS key to be used for encrypting items in the bucket.
```
Comment (shivlaks) [inline packages/aws-cdk/lib/api/bootstrap-environment.ts]: by existing, do we mean a user managed KMS key?

File: packages/aws-cdk/lib/api/bootstrap-environment.ts (hunk 1 above)
Comment (shivlaks) [reply packages/aws-cdk/lib/api/bootstrap-environment.ts]: right, you'd want to default that to a KMS managed server-side encryption

**Comment by hoegertn** [comment]
@eladb 	please close, as #3634 is merged

**Pull request closed without merging**